nullable.go

```diff
@@ -0,0 +1,39 @@
+package tfe
+
+import "encoding/json"
+
+// Nullable acts as a wrapper to enable using "omitempty" on fields that should retain the ability to explicitly be set
+// to null.
+//
+// This is particularly useful for attributes that are set to null if provided as null in a json request body.
+// Conversely, if they are omitted from the request body, they will retain their existing value.
+//
+// This struct should use generics, but this project doesn't support them yet :(
+type Nullable struct {
+	value interface{}
+}
```

Comment:
All the research I did into this problem was pretty fruitless. The only helpful thing I learned is that the concept of an `Optional` is very "un-go-like", which certainly made me feel less great about this solution.

If anyone has any alternative suggestions, I'd be more than happy to hear them!